mongo/integration/client_test.go

```diff
@@ -414,6 +414,30 @@ func TestClient(t *testing.T) {
 			assert.Equal(mt, testAppName, appName, "expected app name %v at index %d, got %v", testAppName, idx, appName)
 		}
 	})
+	// Test that direct connections work as expected.
```

Comment:
This is an integration test to make sure all of the layers are working correctly. It works because drivers overwrite the read preference from `primary` to `primaryPreferred` when creating a direct connection.